kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -878,26 +878,28 @@ impl<'tcx> GotocCtx<'tcx> {
     /// https://doc.rust-lang.org/std/intrinsics/fn.ptr_offset_from.html
     fn codegen_ptr_offset_from(
         &mut self,
+        intrinsic: &str,
         mut fargs: Vec<Expr>,
         p: &Place<'tcx>,
         loc: Location,
     ) -> Stmt {
-        let a = fargs.remove(0);
-        let b = fargs.remove(0);
-        let pointers_to_same_object = a.clone().same_object(b.clone());
+        let dst_ptr = fargs.remove(0);
+        let src_ptr = fargs.remove(0);
 
-        Stmt::block(
-            vec![
-                self.codegen_assert(
-                    pointers_to_same_object,
-                    PropertyClass::PointerOffset,
-                    "ptr_offset_from: pointers point to same object",
-                    loc.clone(),
-                ),
-                self.codegen_expr_to_place(p, a.sub(b)),
-            ],
-            loc,
-        )
+        // Check that the computation would not overflow an `isize`
+        let cast_dst_ptr = dst_ptr.clone().cast_to(Type::ssize_t());
+        let cast_src_ptr = src_ptr.clone().cast_to(Type::ssize_t());
+        let offset = cast_dst_ptr.sub_overflow(cast_src_ptr);
+        let overflow_check = self.codegen_assert(
+            offset.overflowed.not(),
```

Comment:
does this cause performance problems

Reply:
I couldn't come up with a test for this overflow check and then I realized it's wrong. The overflow check has to be done for the distance in bytes, which is what we do now. An `expected` test for the overflow case is included.